okta/table_okta_password_policy.go

```diff
@@ -52,27 +52,90 @@ func listPolicies(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateDat
 	logger := plugin.Logger(ctx)
 	client, err := Connect(ctx, d)
 
-	input := &query.Params{}
 	if err != nil {
-		logger.Error("listOktaPolicies", "connect_error", err)
+		logger.Error("listPolicies", "connect_error", err)
 		return nil, err
 	}
 
-	input.Expand = "rules"
+	input := &query.Params{}
 	switch d.Table.Name {
 	case "okta_password_policy":
 		input.Type = "PASSWORD"
 	case "okta_mfa_policy":
 		input.Type = "MFA_ENROLL"
 	}
 
+	config := GetConfig(d.Connection)
+	if config.EngineType != nil && *config.EngineType == "identity" {
+		return listOktaPoliciesIdentityEngine(ctx, client, d, input)
+	}
+	return listOktaPoliciesClassicEngine(ctx, client, d, input)
+}
+
+func listOktaPoliciesClassicEngine(ctx context.Context, client *okta.Client, d *plugin.QueryData, input *query.Params) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	input.Expand = "rules"
+
+	policies, resp, err := listPoliciesWithSettings(ctx, *client, input)
+	if err != nil {
+		logger.Error("listOktaPoliciesClassicEngine", "list_policies_with_settings_error", err)
+		return nil, err
+	}
+
+	for _, policy := range policies {
+		d.StreamListItem(ctx, policy)
+
+		// Context can be cancelled due to manual cancellation or the limit has been hit
+		if d.RowsRemaining(ctx) == 0 {
+			return nil, nil
+		}
+	}
+
+	// paging
+	for resp.HasNextPage() {
+		var nextPolicySet []*PolicyStructure
+		resp, err = resp.Next(ctx, &nextPolicySet)
+		if err != nil {
+			logger.Error("listOktaPoliciesClassicEngine", "list_policies_with_settings_paging_error", err)
+			return nil, err
+		}
+		for _, policy := range nextPolicySet {
+			d.StreamListItem(ctx, policy)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+func listOktaPoliciesIdentityEngine(ctx context.Context, client *okta.Client, d *plugin.QueryData, input *query.Params) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
 	policies, resp, err := listPoliciesWithSettings(ctx, *client, input)
 	if err != nil {
-		logger.Error("listPolicies", "list_policies_with_settings_error", err)
+		logger.Error("listOktaPoliciesIdentityEngine", "list_policies_error", err)
 		return nil, err
 	}
 
 	for _, policy := range policies {
+		// Additional API call to get rules for Identity Engine
+		rules, _, err := client.Policy.ListPolicyRules(ctx, policy.Id)
```

Comment:
Could we add a check before making the API call? If the rules are missing, then proceed with the API call; otherwise, skip it.